internal/pkg/service/stream/storage/level/local/diskwriter/network/router/balancer/balancer.go

```diff
@@ -12,12 +12,12 @@ import (
 // If no pipeline is ready, the NoPipelineReadyError is returned.
 // The implementation must provide high throughput.
 type Balancer interface {
-	WriteRecord(c recordctx.Context, pipelines []SlicePipeline) (pipeline.RecordStatus, error)
+	WriteRecord(c recordctx.Context, pipelines []SlicePipeline) (pipeline.RecordStatus, int, error)
```

Comment:
I am not huge fan of more than `2` return values. Can we put the number into `RecordStatus`?

Reply:
RecordStatus here tells whether the record was `accepted` or `processed`.

I take it as a standard Write method + status, I don't mind.

Reply:
For me, these two solutions make sense:
- `(pipeline.RecordStatus, int, error)`
- `(pipeline.WriteResult, error)`, `pipeline.WriteResult`  is struct

Reply:
This is really a detail, I suggest you to call about it after Martin's vacation, and don't block the merge, it can always be adjusted.

Reply:
I would go with `WriteResult` seems better